keras_cv/models/segmentation/segformer/segformer.py

```diff
@@ -0,0 +1,158 @@
+import copy
+
+import tensorflow as tf
+
+from keras_cv.backend import keras
+from keras_cv.models.segmentation.segformer.segformer_presets import (  # noqa: E501
+    presets,
+)
+from keras_cv.models.segmentation.segformer.segformer_presets import (  # noqa: E501
+    presets_with_weights,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+from keras_cv.utils.train import get_feature_extractor
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class SegFormer(Task):
+    """A Keras model implementing the SegFormer architecture for semantic
+    segmentation.
+
+    References:
+        - [SegFormer: Simple and Efficient Design for Semantic Segmentation with Transformers](https://arxiv.org/abs/2105.15203)
+        - [Based on the TensorFlow implementation from DeepVision](https://github.com/DavidLandup0/deepvision/tree/main/deepvision/models/segmentation/segformer)
+
+    Args:
+        backbone: `keras.Model`. The backbone network for the model that is
+            used as a feature extractor for the SegFormer encoder. It is *intended*
+            to be used only with the MiT backbone model which was created specifically
+            for SegFormers. It should either be a `keras_cv.models.backbones.backbone.Backbone` or a
+            `tf.keras.Model` that implements the `pyramid_level_inputs`
+            property with keys "P2", "P3", "P4", and "P5" and layer names as
+            values.
+        num_classes: int, the number of classes for the detection model. Note
+            that the `num_classes` doesn't contain the background class, and the
```

Comment:
We should probably change this docstring. The version in DeepLabV3Plus that this comes from is poorly worded IMO. Users _should_ include a background class, so e.g. for cats vs dogs `num_classes=3`.